.github/workflows/pr.yml

```diff
@@ -10,16 +10,15 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
-      - name: Setup node
-        uses: actions/setup-node@v4
+      - name: Setup Bun
```

Comment:
people will still mainly use npm to work with cli, let's leave in the npm version, we don't want to skip someone accidentally using bun only API in the code

Reply:
Can we go full Bun? Or maybe that's a conversation we need to have offline?

Reply:
What do you mean? What about users who `npm I -g neonctl` (I assume that's a majority)

Reply:
Yeah, you're right. We should continue, for now, to run the tests using Node.js as the runtime as most people will run `npm I -g neonctl`.

But, when we adopt https://bun.sh/docs/bundler/executables, we can go full Bun and even use Bun-only APIs.

Reply:
but why?

Reply:
It should be faster to execute than using the Node.js runtime.

Reply:
My thought process is:

- While Bun aims to have complete Node.js compatibility - in practice there's incompatibilities that may not let us fully assume this.
- Bun's single-executable sounds like a very promising replacement for `pkg` and the way we publish standalone binaries (brew & gh releases).
- If we use the above single-executable - we'd effectively use Bun's runtime for binaries and Node.js for `npm i -g neonctl`, due to the 1st point - I'd feel more confident if we only test & support 1 runtime. So maybe it'd be good for us to make `npm i -g neonctl` use the Bun runtime (not super trivial, but I had some ideas)

Reply:
I agree with Amit's thought process here. But before we explore this, should we maybe run our CI on both runtimes to be safe?

Reply:
UPDATE:

I added `setup-node` back (without `cache` cause that's not needed anymore) so that we correctly setup the node version in `.nvmrc`.

I verified, running `bun run test` will try to run `vitest` binary, but since it has `#!/usr/bin/env node` - bun will simply run `node`.

Bun supports overriding shebangs via `bun run --bun test`, which will ignore the shebang and use Bun's runtime, but we're not doing that - so tests are all running as before.

Reply:
but users already have the choice: they can use node via `npm i -g neonctl` or bun via `bun add neonctl`

you think there are no use cases when someone wants to specifically use node version?

Reply:
> you think there are no use cases when someone wants to specifically use node version?

Yeah, it's a CLI, as far as users are concerned it can be natively built with Rust/Go IMO.

Reply:
yea, you have a point